tests/testsuite.src/run_misc.at

```diff
@@ -15251,3 +15289,358 @@ AT_CHECK([$COMPILE prog.cob])
 AT_CHECK([$COBCRUN_DIRECT ./prog], [0], [1.2345E-5 1.2345E-5], [])
 
 AT_CLEANUP
+
+
+# See bug #948 - Comparison with HIGH-VALUE in presence of collating sequences
+AT_SETUP([LOW/HIGH-VALUE when using non-native program collating sequence])
+AT_KEYWORDS([LOW-VALUE HIGH-VALUE ALPHABET EBCDIC ASCII])
+
+AT_DATA([prog1.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       ENVIRONMENT DIVISION.
+       CONFIGURATION SECTION.
+       OBJECT-COMPUTER. x86,
+           PROGRAM COLLATING SEQUENCE IS alpha-custom.
+       SPECIAL-NAMES.
+           ALPHABET alpha-custom IS
+               64 THRU 1
+               65 THRU 192
+               256 THRU 193.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01 LV     PIC X.
+       88 IS-LV  VALUE LOW-VALUE.
+       01 LV-VAL REDEFINES LV BINARY-CHAR UNSIGNED.
+       01 HV     PIC X.
+       88 IS-HV  VALUE HIGH-VALUE.
+       01 HV-VAL REDEFINES HV BINARY-CHAR UNSIGNED.
+       PROCEDURE DIVISION.
+           MOVE LOW-VALUE TO LV.
+           MOVE HIGH-VALUE TO HV.
+           DISPLAY "LOW-VALUE: " LV-VAL
+                 " HIGH-VALUE: " HV-VAL.
+           IF LV = LOW-VALUE AND IS-LV
+             DISPLAY "LOW-VALUE OK"
+           ELSE
+             DISPLAY "LOW-VALUE KO"
+           END-IF.
+           IF HV = HIGH-VALUE AND IS-HV
+             DISPLAY "HIGH-VALUE OK"
+           ELSE
+             DISPLAY "HIGH-VALUE KO"
+           END-IF.
+           IF "X" > LOW-VALUE AND "X" > LV
+              DISPLAY "X > LOW-VALUE"
+           ELSE
+              DISPLAY "X < LOW-VALUE"
+           END-IF.
+           IF "X" < HIGH-VALUE AND "X" < HV
+              DISPLAY "X < HIGH-VALUE"
+           ELSE
+              DISPLAY "X > HIGH-VALUE"
+           END-IF.
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE prog1.cob], [0], [], [])
+AT_CHECK([$COBCRUN_DIRECT ./prog1], [0],
+[LOW-VALUE: 063 HIGH-VALUE: 192
+LOW-VALUE OK
+HIGH-VALUE OK
+X > LOW-VALUE
+X < HIGH-VALUE
+], [])
+
+AT_DATA([prog2.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       ENVIRONMENT DIVISION.
+       CONFIGURATION SECTION.
+       OBJECT-COMPUTER. x86,
+           PROGRAM COLLATING SEQUENCE IS alpha-custom.
+       SPECIAL-NAMES.
+           ALPHABET alpha-custom IS
+               65.
```

Comment:
Please add, possibly in a separate testcase, an example with `ALPHABET ... IS HIGH-VALUE QUOTE SPACE LOW-VALUE` - this should work and use the "native" values - so always x'00' and x'FF'.